### PR TITLE
MCenter action log from mcenter-client

### PR DESCRIPTION
**Pull request by amarmudrankit1**
Basic framework to query action log from mcenter-client

@liororama @yakov-g @sriram109 @harshilpm

### CODE REVIEW

File: reflex-common/reflex-common/src/main/python/mcenter_cli/parallelm/mcenter_cli/list_actions.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import datetime
+
+def list_actions(mclient, args):
+    success = None
+    if args.success:
+        success = True
+    if args.failed:
+        success = False
+    print("Getting action entries for: ")
+    print("    Object: " + str(args.object))
```
Comment (harshilpm) [inline reflex-common/reflex-common/src/main/python/mcenter_cli/parallelm/mcenter_cli/list_actions.py]: you can write as `print("    Object: {}".format(args.object))`

**Review by harshilpm — APPROVED** [review]

File: reflex-common/reflex-common/src/main/python/mcenter_cli/parallelm/mcenter_cli/list_actions.py (hunk 1 above)
Comment (amarmudrankit1) [reply reflex-common/reflex-common/src/main/python/mcenter_cli/parallelm/mcenter_cli/list_actions.py]: Fixed

File: reflex-common/reflex-common/src/main/python/mcenter_cli/parallelm/mcenter_cli/main.py [hunk 2]
```diff
@@ -167,6 +169,13 @@ def _model_upload(mclient, args):
         model_upload_parser.add_argument('name', action='store', help="Model name to use")
         model_upload_parser.add_argument('format', action='store', help="Model format to use")
 
+    def _add_action_log_commands(self):
+        action_log_parser = self._register_command('get-action-log', help='Get Action log for MCenter', action=list_actions)
+        action_log_parser.add_argument('--object', action='store', help='Objects for which to get actions')
+        action_log_parser.add_argument('--action', action='store', help='actions to query')
+        action_log_parser.add_argument('--username', action='store', help='User, whose actions should be queried')
+        action_log_parser.add_argument('--status', action='store', help='Show actions with particular status, [both, success, failed]')
```
Comment (yakov-g) [inline reflex-common/reflex-common/src/main/python/mcenter_cli/parallelm/mcenter_cli/main.py]: parser.add_argument('move', choices=['rock', 'paper', 'scissors'])
thats what you need.
Google for: python option choices

File: reflex-common/reflex-common/src/main/python/mcenter_client_v0/parallelm/mcenter_client/mcenter_client.py [hunk 3]
```diff
@@ -1643,6 +1643,33 @@ def delete_source_control_comp(self, comp_id):
...
+                tuple_list.append(('action', action))
+        if success is not None:
+            tuple_list.append(('success', success))
+        if user_list:
+            for username in user_list:
+                tuple_list.append(('username', username))
+
+        url = self._build_url('actionLog') + "?"
+        url += urlencode(tuple_list)
+        print("URL: " + str(url))
```
Comment (liororama) [inline reflex-common/reflex-common/src/main/python/mcenter_client_v0/parallelm/mcenter_client/mcenter_client.py]: Debug leftover?

File: reflex-common/reflex-common/src/main/python/mcenter_cli/parallelm/mcenter_cli/main.py [hunk 4]
```diff
@@ -167,6 +169,13 @@ def _model_upload(mclient, args):
         model_upload_parser.add_argument('name', action='store', help="Model name to use")
         model_upload_parser.add_argument('format', action='store', help="Model format to use")
 
+    def _add_action_log_commands(self):
```
Comment (liororama) [inline reflex-common/reflex-common/src/main/python/mcenter_cli/parallelm/mcenter_cli/main.py:173]: Add an example(s) usage to examples in the main

File: reflex-common/reflex-common/src/main/python/mcenter_client_v0/parallelm/mcenter_client/mcenter_client.py [hunk 5]
```diff
@@ -1643,6 +1643,33 @@ def delete_source_control_comp(self, comp_id):
         r = self.delete(url, timeout=60)
         return r
 
+    def get_action_log(self, object_list=None, action_list=None, success=None, user_list=None, start_time=None, end_time=None):
```
Comment (liororama) [inline reflex-common/reflex-common/src/main/python/mcenter_client_v0/parallelm/mcenter_client/mcenter_client.py:1646]: Add doc str to method.. with examples of parameters.

**Review by liororama — APPROVED** [review]

File: reflex-common/reflex-common/src/main/python/mcenter_cli/parallelm/mcenter_cli/main.py (hunk 2 above)
Comment (amarmudrankit1) [reply reflex-common/reflex-common/src/main/python/mcenter_cli/parallelm/mcenter_cli/main.py]: Fixed

File: reflex-common/reflex-common/src/main/python/mcenter_client_v0/parallelm/mcenter_client/mcenter_client.py (hunk 3 above)
Comment (amarmudrankit1) [reply reflex-common/reflex-common/src/main/python/mcenter_client_v0/parallelm/mcenter_client/mcenter_client.py]: Fixed

File: reflex-common/reflex-common/src/main/python/mcenter_cli/parallelm/mcenter_cli/main.py (hunk 4 above)
Comment (amarmudrankit1) [reply reflex-common/reflex-common/src/main/python/mcenter_cli/parallelm/mcenter_cli/main.py:173]: Fixed

File: reflex-common/reflex-common/src/main/python/mcenter_client_v0/parallelm/mcenter_client/mcenter_client.py (hunk 5 above)
Comment (amarmudrankit1) [reply reflex-common/reflex-common/src/main/python/mcenter_client_v0/parallelm/mcenter_client/mcenter_client.py:1646]: Fixed

**Review by yakov-g — APPROVED** [review]

**Review by sriram109 — APPROVED** [review]

**Pull request merged**